src/syft/core/node/domain/service/request_message.py

```diff
@@ -207,4 +211,11 @@ def process(node: AbstractNode, msg: RequestMessage, verify_key: VerifyKey) -> N
 
         # using the local arrival time we can expire the request
         msg.set_arrival_time(arrival_time=time.time())
+
+        # At receiving a request from DS, we clear it's object_tags, and re-set it as the
+        # tags of the requested object. Because the DS may give fake tags.
+        while msg.object_tags != []:
```

Comment:
I prefer

```
while len(msg.object_tags):
...
```

Reply:
Yes, your way looks better. I will change it.